ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -25,19 +25,54 @@ open Auth_signature
 
 let net_cmd = !Xapi_globs.net_cmd
 
+let wb_cmd = !Xapi_globs.wb_cmd
+
 let tdb_tool = !Xapi_globs.tdb_tool
 
-let winbind_service = "winbind"
+let debug_level = !Xapi_globs.winbind_debug_level |> string_of_int
 
 module Winbind = struct
+  let name = "winbind"
+
+  let is_ad_enabled ~__context =
+    ( Helpers.get_localhost ~__context |> fun self ->
+      Db.Host.get_external_auth_type ~__context ~self )
+    |> fun x -> x = Xapi_globs.auth_type_AD
+
   let start ~timeout ~wait_until_success =
-    Xapi_systemctl.start ~timeout ~wait_until_success winbind_service
+    Xapi_systemctl.start ~timeout ~wait_until_success name
 
   let stop ~timeout ~wait_until_success =
-    Xapi_systemctl.stop ~timeout ~wait_until_success winbind_service
+    Xapi_systemctl.stop ~timeout ~wait_until_success name
 
   let init_service ~__context =
-    debug "init_service To be implemented in CP-36085"
+    if is_ad_enabled ~__context then
+      start ~wait_until_success:false ~timeout:5.
```

Comment:
You might want to add an `info` log line in the `else` case because you would not call `init_service` is you knew that it would do nothing.